### PR TITLE
[Feature] Support piperider compare between git refs

### PR DESCRIPTION
**Pull request by wcchang1115**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
feature

**What this PR does / why we need it**:
- to support `piperider compare <git-ref>...<git_ref>`

**Which issue(s) this PR fixes**:
sc-31643

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

NONE

### CODE REVIEW

File: piperider_cli/cli_utils/compare_with_recipe.py [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+        else:
+            options['base_ref'] = refs
+
+    if options.get('base_branch') and options.get('base_ref'):
+        return -1, "[bold red]Error:[/bold red] '--base-branch' option and '[REF]' argument cannot be used together"
+
+    return 0, None
+
+
 def compare_with_recipe(**kwargs):
```
Comment (popcornylu) [inline piperider_cli/cli_utils/compare_with_recipe.py:22]: Why not change to use 
```
def compare_with_recipe(refs, **kwargs):
   ...
```
The same as the function defined in cli.py?

**Review by popcornylu — COMMENTED** [review]
Besides the comment in the review, there are some additional notes

1. When we use HEAD, could we mark the HEAD as the current branch in the report?
2. The error message is not clear

    ```
    $ piperider compare main main HEAD
    Error: Commit format is not supported
    ```
    we can say we only support one or no reference
    
3. Clearer help 

     ```
     $ piperider compare --help
     Usage: piperider compare [OPTIONS] [REFS]...
     
     $ git diff --help
     NAME
       git-diff - Show changes between commits, commit and working tree, etc

      SYNOPSIS
         git diff [<options>] [<commit>] [--] [<path>...]
         git diff [<options>] --cached [--merge-base] [<commit>] [--] [<path>...]
         git diff [<options>] [--merge-base] <commit> [<commit>...] <commit> [--] [<path>...]
         git diff [<options>] <commit>...<commit> [--] [<path>...]
   ```
    
image
image

File: piperider_cli/cli_utils/compare_with_recipe.py (hunk 1 above)
Comment (wcchang1115) [reply piperider_cli/cli_utils/compare_with_recipe.py:22]: yes, I can change it.

**Review by popcornylu — APPROVED** [review]
LGTM

**Pull request merged**